test/call.c

```diff
@@ -3066,3 +3066,172 @@ int test_call_hold_resume(void)
+
+	err =  module_load(".", "srtp");
+	err |= module_load(".", "ausine");
+	TEST_ERR(err);
+
+	err = mock_auplay_register(&auplay, baresip_auplayl(),
+		auframe_handler, f);
+	TEST_ERR(err);
+
+	fixture_init_prm(f, ";mediaenc=srtp-mand"
+		";ptime=1;audio_player=mock-auplay,a");
+	mem_deref(f->b.ua);
+	err = ua_alloc(&f->b.ua, "B <sip:b@127.0.0.1>;mediaenc=srtp-mand"
+		";regint=0;ptime=1;audio_player=mock-auplay,b");
+	TEST_ERR(err);
+
+	f->behaviour = BEHAVIOUR_ANSWER;
+	f->estab_action = ACTION_NOTHING;
+
+	/* call established cancel rule */
+	cancel_rule_new(UA_EVENT_CALL_ESTABLISHED, f->a.ua, 0, 0, 1);
+	cancel_rule_and(UA_EVENT_CALL_ESTABLISHED, f->b.ua, 1, 0, 1);
+
+	/* Call A to B */
+	err = ua_connect(f->a.ua, 0, NULL, f->buri, VIDMODE_ON);
+	TEST_ERR(err);
+
+	err = re_main_timeout(5000);
+	TEST_ERR(err);
+	TEST_ERR(fix.err);
+
+	/* verify audio was enabled and bi-directional */
+	ASSERT_TRUE(call_has_audio(ua_call(f->a.ua)));
+	ASSERT_TRUE(call_has_audio(ua_call(f->b.ua)));
+
+	struct sdp_media *m;
+	m = stream_sdpmedia(audio_strm(call_audio(ua_call(f->a.ua))));
+	ASSERT_EQ(SDP_SENDRECV, sdp_media_ldir(m));
+	ASSERT_EQ(SDP_SENDRECV, sdp_media_rdir(m));
+	sdp_media_lattr_apply(m, "crypto", sdp_crypto_handler, &a_tx_key);
+	sdp_media_rattr_apply(m, "crypto", sdp_crypto_handler, &a_rx_key);
+
+	m = stream_sdpmedia(audio_strm(call_audio(ua_call(f->b.ua))));
+	ASSERT_EQ(SDP_SENDRECV, sdp_media_ldir(m));
+	ASSERT_EQ(SDP_SENDRECV, sdp_media_rdir(m));
+	sdp_media_lattr_apply(m, "crypto", sdp_crypto_handler, &b_tx_key);
+	sdp_media_rattr_apply(m, "crypto", sdp_crypto_handler, &b_rx_key);
+
+	/* crosscheck rx & tx keys */
+	TEST_STRCMP(a_rx_key, str_len(a_rx_key), b_tx_key, str_len(b_tx_key));
+	TEST_STRCMP(a_tx_key, str_len(a_tx_key), b_rx_key, str_len(b_rx_key));
+
+	/* rekeying transmission keys from a -> b */
+	re_snprintf(rekeycmd, sizeof(rekeycmd), "srtprekey %s",
+		call_id(ua_call(f->a.ua)));
+	cmd_process_long(baresip_commands(), rekeycmd,
+		str_len(rekeycmd), &pf, NULL);
+
+	cancel_rule_new(UA_EVENT_CUSTOM, f->a.ua, 0, 0, 1);
+	cr->prm = "auframe";
+	cr->n_auframe = 10;
+	cancel_rule_and(UA_EVENT_CUSTOM, f->b.ua, 1, 0, 1);
+	cr->prm = "auframe";
+	cr->n_auframe = 10;
+
+	err = re_main_timeout(5000);
+	TEST_ERR(err);
+	TEST_ERR(fix.err);
+
+	m = stream_sdpmedia(audio_strm(call_audio(ua_call(f->a.ua))));
+	ASSERT_EQ(SDP_SENDRECV, sdp_media_ldir(m));
+	ASSERT_EQ(SDP_SENDRECV, sdp_media_rdir(m));
+	sdp_media_lattr_apply(m, "crypto", sdp_crypto_handler, &a_tx_key_new);
+	sdp_media_rattr_apply(m, "crypto", sdp_crypto_handler, &a_rx_key_new);
+
+	m = stream_sdpmedia(audio_strm(call_audio(ua_call(f->b.ua))));
+	ASSERT_EQ(SDP_SENDRECV, sdp_media_ldir(m));
+	ASSERT_EQ(SDP_SENDRECV, sdp_media_rdir(m));
+	sdp_media_lattr_apply(m, "crypto", sdp_crypto_handler, &b_tx_key_new);
+	sdp_media_rattr_apply(m, "crypto", sdp_crypto_handler, &b_rx_key_new);
+
+	/* transmission key of a must change */
+	ASSERT_TRUE(str_casecmp(a_tx_key, a_tx_key_new));
```

Comment:
I think str_casecmp() returns int ?

Reply:
Thats true and wanted here. We want a `!= 0` value which evaluates to true. A unchanged key would evaluate to `0` and the `ASSERT_TRUE` would fail.
We have no direct macro to check inequality of strings.

Reply:
better to write it like this:

```
ASSERT_TRUE(0 != str_casecmp())
```